src/mon/NVMeofGwSerialize.h

```diff
@@ -308,23 +313,35 @@ inline  void decode(
 }
 
 inline void encode(const NvmeGwClientState& state,  ceph::bufferlist &bl, uint64_t features) {
-  ENCODE_START(1, 1, bl);
+  uint8_t version = 1;
+  if (HAVE_FEATURE(features, NVMEOF_BEACON_DIFF)) {
+     version = 2;
+  }
+  ENCODE_START(version, version, bl);
   encode(state.group_id, bl);
   encode(state.gw_map_epoch, bl);
   encode (state.subsystems, bl, features);
   encode((uint32_t)state.availability, bl);
+  if (version >= 2) {
+    encode((uint64_t)state.last_beacon_seq_number, bl);
+    encode(state.last_beacon_seq_ooo, bl);
+  }
```

Comment:
This encoding a) sets the encode version depending on features and b) sets COMPAT to match version.  I'm not sure you actually need to bother.  You are simply appending two fields, so unconditionally encoding at ENHANCED with LEGACY as COMPAT should be fine.  Is there a specific reason you need to avoid sending ENHANCED to a LEGACY client in this case?